packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -83,6 +142,112 @@ const CTAButton = ({
   )
 }
 
+const PermissionsSection = ({
+  scope,
+  isLoggedIn,
+  userEmail
+}: {
+  scope: string | string[] | null
+  isLoggedIn: boolean
+  userEmail?: string | null
+}) => {
+  return (
+    <>
+      <div className={styles.permsTitleContainer}>
+        <h3 className={styles.infoSectionTitle}>
+          {messages.permissionsRequestedHeader}
+        </h3>
+      </div>
+      <div className={styles.tile}>
+        <div className={styles.permissionContainer}>
+          <div
+            className={cn({
+              [styles.visibilityIconWrapper]: scope === 'read'
+            })}
+          >
+            {scope === 'write' ? (
+              <IconPencil
+                className={cn(styles.permissionIcon)}
+                width={18}
+                height={18}
+              />
+            ) : (
+              <IconVisibilityPublic
+                className={cn(styles.permissionIcon, styles.visibilityIcon)}
+                width={21}
+                height={22}
+              />
+            )}
+          </div>
+
+          <div className={styles.permissionTextContainer}>
+            <span className={styles.permissionText}>
+              {scope === 'write'
+                ? messages.writeAccountAccess
+                : messages.readOnlyAccountAccess}
+            </span>
+            {scope === 'read' ? null : (
+              <div className={cn(styles.permissionDetailTextContainer)}>
+                <p
+                  className={cn(
+                    styles.permissionText,
+                    styles.permissionDetailText
+                  )}
+                >
+                  {messages.doesNotGrantAccessTo}
+                  <br />
+                  {messages.walletsOrDMs}
+                </p>
+              </div>
+            )}
+          </div>
+        </div>
+        <div
+          className={cn(
+            styles.permissionContainer,
+            styles.nonFirstPermissionContainer
+          )}
+        >
+          <div>
+            <IconAtSign
+              width={15}
+              height={15}
+              className={cn(styles.permissionIcon, styles.atSignIcon)}
+            />
+          </div>
+          <div className={styles.permissionTextContainer}>
+            <span className={styles.permissionText}>
+              {messages.emailAddressAccess}
+            </span>
+            {isLoggedIn ? (
+              <div className={cn(styles.permissionDetailTextContainer)}>
+                <span
+                  className={cn(
+                    styles.permissionText,
+                    styles.permissionDetailText,
+                    {
+                      [styles.permissionTextExtraLight]: !userEmail
+                    }
+                  )}
+                >
+                  {userEmail == null ? (
+                    <>
+                      <LoadingSpinner className={styles.loadingSpinner} /> Email
+                      loading...
```

Comment:
should we add to messages object? also, can you use html for the ellipses? not all screenreaders do this correctly: https://www.htmlsymbol.com/html-entities/horizontal-ellipsis.html

Reply:
checkout PublishButton for example

Reply:
ah yes! thanks for catching